internal/pkg/pct/pct.go

```diff
@@ -112,43 +113,40 @@ func (p *Pct) GetInfo(templateCache string, selectedTemplate string) (PuppetCont
 // List lists all templates in a given path and parses their configuration. Does
 // not return any errors from parsing invalid templates, but returns them as
 // debug log events
-func (p *Pct) List(templatePath string, templateName string) ([]PuppetContentTemplate, error) {
+func (p *Pct) List(templatePath string, templateName string) []PuppetContentTemplate {
 	log.Debug().Msgf("Searching %+v for templates", templatePath)
-
-	var matches []string
-
-	// recurse over the templatePath
-	err := p.AFS.Walk(templatePath,
-		func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-
-			// If we find a config file, add it to our matches
-			if match, _ := regexp.MatchString(".*(/|\\\\)"+TemplateConfigFileName+"$", path); match {
-				matches = append(matches, path)
-			}
-
-			return nil
-		})
-
-	if err != nil {
-		return nil, err
-	}
+	// Triple glob to match author/id/version/TemplateConfigFileName
+	// TODO: Make this backward compatible
+	matches, _ := p.IOFS.Glob(templatePath + "/**/**/**/" + TemplateConfigFileName)
 
 	var tmpls []PuppetContentTemplate
 	for _, file := range matches {
 		log.Debug().Msgf("Found: %+v", file)
 		i := p.readTemplateConfig(file).Template
-		tmpls = append(tmpls, i)
+		// Do not write id-less configs (ie, invalid, could not parse) to the return
+		if len(i.Id) > 0 {
+			tmpls = append(tmpls, i)
+		}
+	}
+	// Temporary workaround to find old layout templates
+	oldMatches, _ := p.IOFS.Glob(templatePath + "/**/" + TemplateConfigFileName)
+	for _, file := range oldMatches {
+		log.Debug().Msgf("Found: %+v", file)
+		i := p.readTemplateConfig(file).Template
+		// Do not write id-less configs (ie, invalid, could not parse) to the return
+		if len(i.Id) > 0 {
+			tmpls = append(tmpls, i)
+		}
```

Comment:
Open question: is the only reason this is necessary to support our own tests? or could users have installed templates in the old layout and be broken if we release without this?

Reply:
It is possible that someone writing their own template, using an [older version of the puppet-content-template template](https://github.com/puppetlabs/baker-round/tree/dbb41ddb79a563bc7bbedcc694e4f5b9e7cf14d0/puppet-content-template) (_or manually hand rolling their own, based off the conventions at the time_), could then have this still in their templates directory. If they exported it using `pct build`, then they will get a warning when trying to install it from 

It is possible for anyone using `0.3.0` to have utilised the `pct build` command, which could have packaged up templates in the old layout, but when they go to install them using `>= 0.4.0`, they'll encounter an error (_presumably_) as the template is not in the expected directory layout.

Reply:
It should be to support our own tests prior to 0.4.0 work being complete. 0.4.0 release should mean that we no longer need the workarounds. When we release we are not building in backwards compatibility. We will need to flag this in release notes so that folks who may have already jumped head first into the new system can 'build' and 'install' their template into the correct structure.